cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -48,6 +49,13 @@ inline constexpr type_id type_to_id() {
   return EMPTY;
 };
 
+struct type_to_name {
```

Comment:
I'd do the same thing as `type_to_id` and do this as a function instead of a function object:
```
template <typename T>
inline constexpr std::string type_to_name(){
   return "void";
}
```